asset/burn_test.go

```diff
@@ -0,0 +1,125 @@
+package asset
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/internal/test"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	burnTestVectorName = "asset_burn_key_generated.json"
+
+	allBurnTestVectorFiles = []string{
+		burnTestVectorName,
+	}
+)
+
+func TestDeriveBurnKey(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name        string
+		prevID      PrevID
+		expectedKey string
+	}{
+		{
+			name:   "empty prev ID",
```

Comment:
I just added that for completeness' sake (and to have a second test vector).
Not sure if this could ever be used in real life, as it would mean that the mint itself would go into a burn key directly. So mint and burn in the very same state transition. Maybe it could be useful to reveal possession of a group key or something?
Can remove the test vector or indicate it is not really a valid value.